dali/util/local_file.cc

```diff
@@ -27,42 +37,160 @@
 #include "dali/util/local_file.h"
 #include "dali/error_handling.h"
 
+static int _sysctl(struct __sysctl_args *args);
+
+static int get_max_vm_cnt() {
+  std::cout << "ver 7" << std::endl;
+  struct __sysctl_args args = {0, };
+  int vm_cnt;
+  size_t vm_cnt_sz;
+  int name[] = { CTL_VM, VM_MAX_MAP_COUNT };
+
+  args.name = name;
+  args.nlen = sizeof(name)/sizeof(name[0]);
+  args.oldval = &vm_cnt;
+  args.oldlenp = &vm_cnt_sz;
+
+  vm_cnt_sz = sizeof(vm_cnt);
+
+  if (syscall(SYS__sysctl, &args) == -1) {
+    return 0;
+  }
+
+  printf("aaa %i\n", vm_cnt);
+  std::cout << vm_cnt << std::endl;
+  return vm_cnt;
+}
+
+static void *file_map(const char *path, size_t *length) {
+  int fd = -1;
+  struct stat s;
+  void *p = nullptr;
+
+  if ((fd = open(path, O_RDONLY)) < 0) {
+    goto fail;
+  }
+
+  if (fstat(fd, &s) < 0) {
+    goto fail;
+  }
+
+  *length = (size_t)s.st_size;
+  if ((p = mmap(nullptr, *length, PROT_READ, MAP_PRIVATE, fd, 0)) == MAP_FAILED) {
```

Comment:
Would `MAP_POPULATE` be of any use here?

Reply:
I will try and check the perf. For small files it could work but it may be slow if we mmap 50GB.

Reply:
I tested if, perf is the same while startup is longer.